chain-db-watcher/src/index.ts

```diff
@@ -1,70 +1,159 @@
+	const proposalSubscriptionClient = createSubscriptionObservable(
 		graphQLEndpoint,
-		proposalsSubscription
+		proposalSubscription
+	);
+
+	const referendumSubscriptionClient = createSubscriptionObservable(
+		graphQLEndpoint,
+		referendumSubscription
 	);
 
 	console.log(`🚀 Chain-db watcher listening to ${graphQLEndpoint}`);
 
-	subscriptionClient.subscribe(({ data }) => {
-		console.log('Received event: ');
-		console.log(JSON.stringify(data, null, 2));
+	proposalSubscriptionClient.subscribe(
+		({ data }) => {
+			// console.log(`Received Proposal event: ${JSON.stringify(data, null, 2)}`);
 
-		if (data?.proposal.mutation === 'CREATED'){
-			const { proposalId, proposer } = data.proposal.node;
+			if (data?.proposal.mutation === 'CREATED') {
+				const { proposalId, author } = data.proposal.node;
 
-			proposalAlreadyExists(proposalId)
-				.then((alreadyExist) => {
-					if (!alreadyExist) {
-						addPostAndProposal({
-							onchain_proposal_id: proposalId,
-							proposer
-						})
-							.then( () => console.log(`✅ Proposal ${proposalId.toString()} added to the database.`))
-							.catch( error => console.error(`Error adding a new proposal: ${error}`));
-					} else {
-						console.error(`🔴 proposal id ${proposalId.toString()} already exists in the database. Not inserted.`);
+				proposalDiscussionExists(proposalId)
+					.then(alreadyExist => {
+						if (!alreadyExist) {
+							addPostAndProposal({
+								onchainProposalId: proposalId,
+								proposer: author
+							})
+								.then(() =>
+									console.log(
+										`${chalk.green('✔︎')} Proposal ${proposalId.toString()} added to the database.`
+									)
+								)
+								.catch(error =>
+									console.error(
+										chalk.red(`⚠︎ Error adding a new proposal: ${error}`))
+								);
+						} else {
+							console.error(
+								chalk.red(
+									`✖︎ Proposal id ${proposalId.toString()} already exists in the dicsussion database. Not inserted.`
+								)
+							);
+						}
+					})
+					.catch(error => console.error(chalk.red(error)));
+			}
+		},
+		err => {
+			console.error(chalk.red(err));
+		}
+	);
+
+	referendumSubscriptionClient.subscribe(({ data }): void => {
+		// console.log(`Received Referendum event: ${JSON.stringify(data, null, 2)}`);
+
+		if (data?.referendum.mutation === 'CREATED') {
+			const {
+				primage,
+				referendumId,
+				referendumStatus
+			} = data.referendum.node;
+
+			// At referendum creation, there should be only
+			// a "Started" status event.
+			if (!(referendumStatus[0].status === 'Started')) {
+				console.error(
+					chalk.red(
+						`Referendem with id ${referendumId.toString()} has an unexpected status. Expect "Started", got ${referendumStatus[0].status}."`
+					)
+				);
+			}
+			const referendumCreationBlockHash =
+				referendumStatus[0].blockNumber.hash;
+
+			getAssociatedProposalId({
+				preimageHash: primage?.hash,
+				referendumCreationBlockHash
+			})
+				.then(associatedProposalId => {
```

Comment:
.then in my opinion can be moved to async await.

Reply:
I don't think you can do that in a subscribe. `Promise returned in function argument where a void return was expected.` because the subscribe can't be an async function.

Reply:
Unless there's a good reason to move this to another function (which would then allow to use async/await), I'd keep it that way.